From 6dadeaa180878af84243810720ea6a68434507b2 Mon Sep 17 00:00:00 2001
From: Shayne Chen <shayne.chen@mediatek.com>
Date: Wed, 30 Jul 2025 18:00:43 +0800
Subject: [PATCH 009/115] backport-include: add faux.h

This is added from kernel 6.14. Wrap some functions to use back
platform_device API.

Signed-off-by: Shayne Chen <shayne.chen@mediatek.com>
---
 backport-include/linux/device/faux.h | 36 ++++++++++++++++++++++++++++
 1 file changed, 36 insertions(+)
 create mode 100644 backport-include/linux/device/faux.h

diff --git a/backport-include/linux/device/faux.h b/backport-include/linux/device/faux.h
new file mode 100644
index 00000000..574276e6
--- /dev/null
+++ b/backport-include/linux/device/faux.h
@@ -0,0 +1,36 @@
+#ifndef __BACKPORT_DEVICE_FAUX_H
+#define __BACKPORT_DEVICE_FAUX_H
+#include <linux/version.h>
+#if LINUX_VERSION_IS_GEQ(6,14,0)
+#include_next <linux/device/faux.h>
+
+#else
+#include_next <linux/platform_device.h>
+
+#ifndef faux_device
+#define faux_device platform_device
+#endif
+
+#ifndef faux_device_destroy
+static inline
+void backport_faux_device_destroy(struct faux_device *faux_dev)
+{
+	platform_device_unregister(faux_dev);
+}
+#define faux_device_destroy LINUX_BACKPORT(faux_device_destroy)
+#endif /* !faux_device_destroy */
+
+#ifndef faux_device_create
+static inline
+struct faux_device *backport_faux_device_create(const char *name,
+					        struct device *parent,
+						const void *faux_ops)
+{
+	return platform_device_register_simple(name, 0, NULL, 0);
+}
+#define faux_device_create LINUX_BACKPORT(faux_device_create)
+#endif /* !faux_device_create */
+
+#endif /* >= 6.14 */
+
+#endif
-- 
2.45.2

